main.cpp

```diff
@@ -115,18 +121,42 @@ int main()
     /*
      construct 2 more Car instances via emplace_back.
      */
-    
+    cars.emplace_back("Wiiliam");
+    cars.emplace_back("Jose");
     /*
      now reserve and emplace_back your Trucks and Motorcyles
      */
     
+    motorcycles.reserve(3);
+    motorcycles.emplace_back("Jerry");
+    motorcycles.emplace_back("Bill");
+    motorcycles.emplace_back("John");
+
+    trucks.reserve(3);
+    trucks.emplace_back("Nancy");
+    trucks.emplace_back("Ria");
+    trucks.emplace_back("Peace");
     
-    
-    
-    assert(false);
+    //assert(false);
     //add the cars, motorcycles and trucks to the highway using range-based for() loops: for( element : vec ) { ... }
     //be careful to not accidentally make element copies when iterating.
     
+    for (Car& car : cars)
+    {
+        highway.addVehicle(&car);
+    }
+    
+    for (Motorcycle& motorc : motorcycles)
+    {
+        highway.addVehicle(&motorc);
+    }
+
+    for (SemiTruck& semit : trucks)
+    {
+        highway.addVehicle(&semit);
+    }
```

Comment:
use `auto&` in these types of loops.  If you ever change your vector's held type, you won't need to revise your loops.
- [ ] revise